bundler/lib/bundler/gem_helpers.rb

```diff
@@ -48,6 +48,13 @@ def select_best_platform_match(specs, platform)
     end
     module_function :select_best_platform_match
 
+    def force_ruby_platform(specs)
+      matching = specs.select {|spec| spec.force_ruby_platform = true if spec.match_platform(Gem::Platform::RUBY) }
```

Comment:
using a `select` here where the return value is coming from a setter feels a bit non-obvious to me -- maybe expand to

```
select do |spec|
  next unless spec.match_platform(Gem::Platform::RUBY)
  spec.force_ruby_platform = true
  true
end
```

Reply:
Agreed, it was feeling too cryptic when I was writing it. I like your approach better, will switch to it.

Reply:
I'm changing this but now it feels pretty neat 😆, let's ask for a third opinion. @martinemde?

Reply:
Oooh, count me in!

The more I look at it though, the more I lean towards:

```ruby
select do |spec|
  spec.match_platform(Gem::Platform::RUBY) && spec.force_ruby_platform = true
end
```

What I like about it is that it preserves this as a "positive" case match which is what select is all about. "Select specs that match platform ruby && where force_ruby_platform = true returns true".

Select should contain a logical boolean operation in the block, so this preserves that characteristic.

Reply:
Love it!

Reply:
That does have the potential confusion of `==` vs `=`, so maybe RuboCop will complain, let's see.

Reply:
Ugg, RuboCop does not complain but I'm on the fence again because of the potential assignment vs comparison confusion 😅.

Reply:
That's fair. I wish there was a declarative `force_ruby_platform!` (and I would make it only apply to matching ruby platforms).

I like both Sam's and your solutions just fine though. I'm always game to debate code styles but any of them will be fine. The most verbose and intentional is maybe this, but it's inefficient.:

```ruby
specs
  .select {|spec| spec.match_platform(Gem::Platform::RUBY) }
  .each {|spec| spec.force_ruby_platform = true }
```

Absolutely no confusion there, but it's a trade-off for sure.

Reply:
Just add another checkmark to *bundler should ship with rubygems and be upgraded in lockstep* because all these little compromises would go away when you can change rubygems code as needed.

Reply:
oh my, I didn't realize the context here. This is for spec support and no code uses this yet. I'd just be verbose here with the clearest one since it runs once, haha.

Reply:
Mmmm, no this is not only for specs, it's part of the bug fix in #6936. I just took [your suggestion](https://github.com/rubygems/rubygems/pull/6936#discussion_r1314320125) from the original PR since I liked it :)

I went with your declarative idea in the end.